### PR TITLE
chore(chat): Add chat soft return (for #308)

### LINKED ISSUE #308: Chat - Add soft return into messaging
**Task:** In old Uplink we had the ability to Soft Return (Multi-line support) Right now if a user tries to start a new line it will send whatever they have currently typed. 

**OS:**
-Windows
-MacOS

**STR:** 
1. Type message 
2. Hit _Shift+Enter__ to start a new line 

**Screen recording:**

https://user-images.githubusercontent.com/93608357/219474071-51849ecc-5657-4f81-b1d5-2a1242b8f2fc.mov

### PR DESCRIPTION
**Pull request by Flemmli97**
### What this PR does 📖

- While holding shift and pressing enter adds a new line instead of sending the message
- Also fixes chat message not able to show new lines

### Which issue(s) this PR fixes 🔨

- Resolve #308

### CODE REVIEW

**Comment by phillsatellite** [comment]
Found a few things when testing but super neat so far! 🎉 

1. If you send a message with multiple lines, the size of the textbox will persist after sending message 

2. After you leave chat page and come back soft return seems to stop working 

https://user-images.githubusercontent.com/93608357/221042492-480cd5ee-15cb-4ed1-baf3-2afc064df068.mov

File: kit/src/elements/input/mod.rs [hunk 1]
```diff
@@ -304,8 +309,14 @@ pub fn Input<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
         .and_then(|b| b.then_some("password"))
         .unwrap_or("text");
 
+    let multiline =
+        cx.props.allow_line_breaks.unwrap_or(false) && !cx.props.is_password.unwrap_or(false);
```
Comment (dariusc93) [inline kit/src/elements/input/mod.rs]: A bool, by default, would be `false` so unless those would be `true` by default (thus requiring an `unwrap_or` or similar to be set to `true` if those fields are `None`), its best to use `unwrap_or_default`

```suggestion
        cx.props.allow_line_breaks.unwrap_or_default() && !cx.props.is_password.unwrap_or_default();
```

**Review by dariusc93 — DISMISSED** [review]
Looks good to me from a glance over. I did leave a small suggestion but not non-breaking to the PR

**Comment by phillsatellite** [comment]
Not sure if anyone else is able to replicate or not but I got 2 small errors when trying to pull the latest changes, tried clearing cache and rebuilding in case that might have been the problem but still no luck 😕

Screenshot 2023-02-27 at 11 49 05 AM

**Comment by lgmarchi** [comment]
> Not sure if anyone else is able to replicate or not but I got 2 small errors when trying to pull the latest changes, tried clearing cache and rebuilding in case that might have been the problem but still no luck 😕
> 
> Screenshot 2023-02-27 at 11 49 05 AM

I am having same problem as you @phillsatellite

**Review by lgmarchi — CHANGES_REQUESTED** [review]
@Flemmli97 Merge last dev into your branch, so it can fix compile problem libp2p update and allow to test your changes. 

Dev was merged

**Review by lgmarchi — COMMENTED** [review]
Branch dev was merged and fixed libp2p problem when compile

**Comment by phillsatellite** [comment]
Thanks for merging dev @lgmarchi! We actually tested together and found a few issues, we couldnt get soft return to work at all in chat, and when we paste DID on Friends page or we rename a file on Files page there will be a horizontal scrollbar that appears 

image

https://user-images.githubusercontent.com/93608357/221652642-786df649-41db-4fca-a337-db3a0216d751.mov

**Comment by Flemmli97** [comment]
> 1. If you send a message with multiple lines, the size of the textbox will persist after sending message
> 2. After you leave chat page and come back soft return seems to stop working

should be fixed now

**Comment by Flemmli97** [comment]
one more thing: you cant horizontal drag the text if the text is too long.

i tried in a clean dioxus project using a textarea and by default you are unable to drag both vertically and horizontally which differs from what we have (vertical dragging works). idk whats different that allows it

https://user-images.githubusercontent.com/34157027/221722093-023e6c0a-bf36-4e8e-a6dc-1054d5c49f68.mp4

**Review by phillsatellite — DISMISSED** [review]

**Review by lgmarchi — DISMISSED** [review]

**Review by dariusc93 — APPROVED** [review]

**Comment by dariusc93** [comment]
Merging since it was previously approved with the only change was merging in dev (and resolving conflicts related to the imports)

**Pull request merged**